Analysis/tasks.py

```diff
@@ -396,3 +396,100 @@ def run(self):
             with tmp_outFile.localize("r") as tmpFile, self.output().localize("w") as outFile:
                 RenameHistsProducer_cmd = ['python3', RenameHistsProducer,'--inFile', tmpFile.path, '--outFile', outFile.path, '--var', var, '--year', getYear(self.period), '--ana_path', self.ana_path(), '--period', self.period]
                 ps_call(RenameHistsProducer_cmd,verbose=1)
+
+class PlotTask(Task, HTCondorWorkflow, law.LocalWorkflow):
+    max_runtime = copy_param(HTCondorWorkflow.max_runtime, 2.0)
+    n_cpus      = copy_param(HTCondorWorkflow.n_cpus, 1)
+
+    def workflow_requires(self):        
+        merge_map = MergeTask.req(self, branch=-1, branches=(), customisations=self.customisations).create_branch_map()
+        return {"merge": MergeTask.req(self,branches=tuple(merge_map.keys()),customisations=self.customisations,)}
+    
+    def create_branch_map(self):
+        branches = {}
+        merge_map = MergeTask.req(self, branch=-1, branches=(), customisations=self.customisations).create_branch_map()
+
+        for k, (_, (var, _)) in enumerate(merge_map.items()):
+            branches[k] = var
+        return branches
+
+    def requires(self):
+        var = self.branch_data
+
+        merge_map = MergeTask.req(self, branch=-1, branches=(), customisations=self.customisations).create_branch_map()
+        merge_branch = next(br for br, (v, _) in merge_map.items() if v == var)
+
+        return MergeTask.req(self,branch=merge_branch,customisations=self.customisations,max_runtime=MergeTask.max_runtime._default,)
+
+    def output(self):
+        var  = self.branch_data
+        flag_file= os.path.join(self.version, self.period, "plots", var, ".done")
```

Comment:
I don't think flag_file is needed. Please use the expected PDF outputs, instead of making names in `run`. For example, at some point we can have another task that deploys plot to the web page, which will depend on `PlotTask` and will need to know the location of pdfs.